### PR TITLE
Add TRPO policy

### LINKED ISSUE #190: TRPO policy support
Is there a plan to support TRPO algorithm?

### LINKED ISSUE #338: Plans of implementing 3 classic model-free algorithm (TRPO/TNPG/ACKTR), and benchmarking them in mujoco settings
## Purpose
The purpose of this issue(discussion) is to introduce a series of prs in the near future targeted to releasing tianshou's full benchmark for MuJoCo Gym task suite.
This benchmark will include 9 algorithms, 6 of which are already supported by #274, #307, the rest 3 algorithms are currently not supported by Tianshou officially. Because of their popularity, we decided to support/benchmark them.
## Plans
About 6 prs will be introdued in the next 2 weeks(hopefully). 2 prs for each algorithm: one implementing the algorithm and one benchmarking it.

Step 1 is TRPO support, which has already been proposed by #337.

### PR DESCRIPTION
**Pull request by ChenDRAG**
Only draft, open for discussion.

### CODE REVIEW

**Review by Trinkle23897 — DISMISSED** [review]

File: tianshou/policy/modelfree/trpo.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    :param torch.nn.Module actor: the actor network following the rules in
+        :class:`~tianshou.policy.BasePolicy`. (s -> logits)
+    :param torch.nn.Module critic: the critic network. (s -> V(s))
+    :param torch.optim.Optimizer optim: the optimizer for actor and critic network.
+    :param dist_fn: distribution class for computing the action.
+    :type dist_fn: Type[torch.distributions.Distribution]
+    :param bool advantage_normalization: whether to do per mini-batch advantage
+        normalization. Default to True.
+
+    TODO: doc
```
Comment (Trinkle23897) [inline tianshou/policy/modelfree/trpo.py]: @ChenDRAG

File: tianshou/policy/modelfree/trpo.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+                        # calculate kl and if in bound, loss actually down
+                        new_dist = self(b).dist
+                        new_dratio = (
+                            new_dist.log_prob(b.act) - b.logp_old).exp().float()
+                        new_dratio = new_dratio.reshape(
+                            new_dratio.size(0), -1).transpose(0, 1)
+                        new_actor_loss = -(new_dratio * b.adv).mean()
+                        kl = kl_divergence(old_dist, new_dist).mean()
+
+                        # TODO: document 1.5 here
```
Comment (Trinkle23897) [inline tianshou/policy/modelfree/trpo.py]: @ChenDRAG

File: tianshou/policy/modelfree/trpo.py (hunk 1 above)
Comment (ChenDRAG) [reply tianshou/policy/modelfree/trpo.py]: done

**Comment by ChenDRAG** [comment]
Finished I think

**Review by Trinkle23897 — APPROVED** [review]

**Pull request merged**